ibc-core/ics23-commitment/types/src/specs.rs

```diff
@@ -119,16 +138,29 @@ impl From<LeafOp> for RawLeafOp {
 #[derive(Clone, Debug, PartialEq)]
 struct InnerSpec(RawInnerSpec);
 
-impl From<RawInnerSpec> for InnerSpec {
-    fn from(inner_spec: RawInnerSpec) -> Self {
-        Self(RawInnerSpec {
+impl TryFrom<RawInnerSpec> for InnerSpec {
+    type Error = CommitmentError;
+    fn try_from(inner_spec: RawInnerSpec) -> Result<Self, CommitmentError> {
+        if inner_spec.child_size <= 0 {
+            return Err(CommitmentError::InvalidChildSize(inner_spec.child_size));
+        }
+        if inner_spec.min_prefix_length > inner_spec.max_prefix_length
```

Comment:
```suggestion
        if inner_spec.max_prefix_length < inner_spec.min_prefix_length
```

just being pedantic for order consistency.